src/idl_parser.cpp

```diff
@@ -671,6 +680,10 @@ CheckedError Parser::ParseField(StructDef &struct_def) {
   if (LookupCreateStruct(name, false, false))
     return Error("field name can not be the same as table/struct name");
 
+  if (!IsLowerSnakeCase(name)) {
+    Warning("field names SHOULD be lowercase snake_case. Got: " + name);
```

Comment:
no reason to upper-case SHOULD here? And make `.` a `,`

Reply:
I was just using the MUST, SHOULD, MAY style of (https://www.ietf.org/rfc/rfc2119.txt). I changed it to lowercase and fixed the comma.

Reply:
Ahh.. I guess we don't have that style anywhere else yet..